### ISSUE TITLE
Got 401 error in get_limit and get request, and "CAPTCHA verification is required" in custom_generation

### ISSUE THREAD

**Issue by 2easyjunmin** [gcui-art/suno-api#201]
Does anyonen have a solution?

<img width="1524" alt="Screenshot 2024-11-25 at 4 33 03 PM" src="https://github.com/user-attachments/assets/710928ac-d340-482c-9e54-ac8a36b3a9f0">

**Comment by Oozankilic** [comment]
You need to create a song manually from website to pass Captcha verification but the other error has started today, I am working on it

**Comment by 2easyjunmin** [comment]
@Oozankilic 
Thanks for your fast reply.
I don't get the captcha when I go to the website, it works fine on the website, and I think the website keeps changing cookies.
I've updated them all and it doesn't work.
How long will it take?

**Comment by bhushanrathod32** [comment]
private async getClerkLatestVersion() {
    ...
    // Define the Clerk version as a constant or fetch it dynamically
    const STABLE_CLERK_VERSION = "4.74.0";
    this.clerkVersion = STABLE_CLERK_VERSION;
}


ensure it uses a Clerk version released prior to the implementation of fraud detection

**Comment by teal33t** [comment]
> private async getClerkLatestVersion() { ... // Define the Clerk version as a constant or fetch it dynamically const STABLE_CLERK_VERSION = "4.74.0"; this.clerkVersion = STABLE_CLERK_VERSION; }
> 
> ensure it uses a Clerk version released prior to the implementation of fraud detection

This works!

**Comment by 2easyjunmin** [comment]
Is it still essential to periodically make manual requests on the web?
When do cookies expire and need to be updated?

@bhushanrathod32 @teal33t

**Comment by bhushanrathod32** [comment]
Normally expiry time for cookies is 400 days
You can find cookie expiry by this link https://clerk.suno.com/v1/client?_clerk_js_version=5.35.0

For hcaptcha you have to make manual request on create page when hcaptcha token expires

**Comment by Kev2012** [comment]
How long is the time from expire Hcaptcha? @bhushanrathod32 ?

**Comment by bhushanrathod32** [comment]
https://github.com/gcui-art/suno-api/issues/198#issuecomment-2495421159

Probably 2hr

**Comment by 2easyjunmin** [comment]
Do you have ideas for automating manual requests?
I don't think it's possible to check every 2 hours, especially in the wee hours of the morning :) @bhushanrathod32

**Comment by bhushanrathod32** [comment]
At the moment I don't have any idea, but looking for solutions. Will surely update here.

Some captcha solving tools might help, someone in another issue posted about cap.guru

**Comment by stenszkysoma** [comment]
hey, will this be fixed? I would love to use it further. let me know if you need assistance, perhaps we as community can figure out together.

**Comment by bhushanrathod32** [comment]
Found a solution on another thread hope it works https://github.com/gcui-art/suno-api/pull/193#issuecomment-2490588270
Possible solution https://github.com/gcui-art/suno-api/pull/193#issuecomment-2485759944

**Comment by Germey** [comment]
Actually SUNO hCaptcha detected not only the captcha result, but also detect the mouse path, clicking behavior, etc.  The crack result token contains all of the behavior infos. And SUNO can know if you are a real human according to the final hCaptcha crack token.

So, the only way is you need to totally simulate the human behavior of hCaptcha using browser, otherwise you will always get "This is a fake app....", even though you cracked it.

I have successfully cracked using above solution, and now it works well now, and the result does not contains "This is a fake app anymore".

Also I exposed and provided above API for suno, but paid, just FYI if you are interested: 

* API: https://platform.acedata.cloud/services/suno
* Doc: https://platform.acedata.cloud/documents/suno-audios-integration

**Comment by glitchbunny0** [comment]
The 401 + captcha-required errors are fixed in [my fork](https://github.com/glitchbunny0/suno-api) (PR #286). The root cause was two-fold: (1) session keep-alive was failing on stale cookies — fixed with proper Clerk version detection; (2) generation is now gated by captcha (hCaptcha v1 or Turnstile v2, auto-detected from `/api/c/check`). The fork integrates 2Captcha to solve both. Also discovered captcha is risk-based — it only triggers under rapid-fire usage, not on every generation.
